web/app/controllers/login.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import Ember from 'ember';
+
+
+export default Ember.Controller.extend({
+  session: Ember.inject.service(),
+  notifications: Ember.inject.service('notification-messages'),
+  errorMessage: '',
+  showError: false,
+  schedulerUrl: null,
+  cluster: null,
+  actions: {
+    login() {
+      let { username, password } = this.getProperties('username', 'password');
+      username = this.getValidStringValue(username);
+      password = this.getValidStringValue(password);
+      if (!username.length || !password.length) {
+        this.set("showError", true);
+        this.set("errorMessage", "Username or Password cannot be empty");
+      } else if (this.cluster === null || this.schedulerUrl === null){
+        /*if cluster or the schedulerUrl is empty this means that
+           login route is not transitioned from the job page */
+        this.set("showError", true);
+        this.set("errorMessage", "Cannot process your request!! Visit the respective Job Page and retry");
```

Comment:
As these two lines are recurring, it would be better to isolate it for modularity.